app/models/report.rb

```diff
@@ -51,6 +51,14 @@ def destroy_reported_item
     mark_as_reviewed
   end
 
+  def mark_as_nsfw
```

Comment:
These changes will not federate, so the post will still be SFW everywhere else. I don't think that this is a good thing to implement then...